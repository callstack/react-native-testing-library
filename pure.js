// makes it so people can import from '@testing-library/react-native/pure'
module.exports = require('./build/pure');
