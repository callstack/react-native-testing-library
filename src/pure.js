// @flow
import act from './act';
import cleanup from './cleanup';
import fireEvent from './fireEvent';
import flushMicrotasksQueue from './flushMicroTasks';
import render from './render';
import shallow from './shallow';
import waitFor, { waitForElement } from './waitFor';
import waitForElementToBeRemoved from './waitForElementToBeRemoved';
import { within, getQueriesForElement } from './within';

export { act };
export { cleanup };
export { fireEvent };
export { flushMicrotasksQueue };
export { render };
export { shallow };
export { waitFor, waitForElement };
export { waitForElementToBeRemoved };
export { within, getQueriesForElement };
