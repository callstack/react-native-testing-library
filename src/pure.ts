import act from './act';
import cleanup from './cleanup';
import fireEvent from './fireEvent';
import render from './render';
import waitFor from './waitFor';
import waitForElementToBeRemoved from './waitForElementToBeRemoved';
import { within, getQueriesForElement } from './within';
import { getDefaultNormalizer } from './matches';
import { renderHook } from './renderHook';
import { screen } from './screen';
import { isInaccessible } from './helpers/accessiblity';

export type {
  RenderOptions,
  RenderResult,
  RenderResult as RenderAPI,
} from './render';
export type { RenderHookOptions, RenderHookResult } from './renderHook';

export { act };
export { cleanup };
export { fireEvent };
export { render };
export { waitFor };
export { waitForElementToBeRemoved };
export { within, getQueriesForElement };
export { getDefaultNormalizer };
export { renderHook };
export { screen };
export { isInaccessible };
