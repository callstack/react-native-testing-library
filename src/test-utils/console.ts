import { format } from 'util';

export function excludeConsoleMessage(logFn: (...args: unknown[]) => void, excludeMessage: string) {
  return (...args: unknown[]) => {
    const message = format(...args);
    if (message.includes(excludeMessage)) {
      return;
    }

    logFn(...args);
  };
}
