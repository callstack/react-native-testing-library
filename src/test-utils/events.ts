interface EventEntry {
  name: string;
  payload: any;
}

export function createEventLogger() {
  const events: EventEntry[] = [];
  const logEvent = (name: string) => {
    return (event: unknown) => {
      const eventEntry: EventEntry = {
        name,
        payload: event,
      };

      events.push(eventEntry);
    };
  };

  return { events, logEvent };
}
