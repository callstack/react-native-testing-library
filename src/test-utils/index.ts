export * from './console';
export * from './events';
