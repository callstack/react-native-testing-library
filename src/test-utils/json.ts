import type { ReactTestRendererJSON } from 'react-test-renderer';

type JsonPropsMapper = {
  [key: string]: unknown;
};

export function mapJsonProps<T extends ReactTestRendererJSON | ReactTestRendererJSON[] | null>(
  element: T,
  mapper: JsonPropsMapper,
): T {
  if (Array.isArray(element)) {
    return element.map((e) => mapJsonProps(e, mapper)) as T;
  }

  if (!element) {
    return element;
  }

  const resultProps = { ...element.props };
  Object.keys(mapper).forEach((key) => {
    if (key in element.props) {
      resultProps[key] = mapper[key];
    }
  });

  const resultElement = { ...element, props: resultProps };
  Object.defineProperty(resultElement, '$$typeof', {
    value: Symbol.for('react.test.json'),
  });

  return resultElement;
}
