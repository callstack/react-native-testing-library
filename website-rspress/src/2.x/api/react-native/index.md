# Re.Pack React Native APIs

## Table of contents

### Classes

- [Chunk](./classes/Chunk.md)
- [ChunkManager](./classes/ChunkManager.md)

### Interfaces

- [ChunkConfig](./interfaces/ChunkConfig.md)
- [ChunkManagerConfig](./interfaces/ChunkManagerConfig.md)
- [RemoteChunkLocation](./interfaces/RemoteChunkLocation.md)
- [StorageApi](./interfaces/StorageApi.md)

### Type aliases

- [RemoteChunkResolver](./types/RemoteChunkResolver.md)
