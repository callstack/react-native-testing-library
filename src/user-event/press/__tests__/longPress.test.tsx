import React from 'react';
import { Pressable, Text } from 'react-native';
import { render, screen } from '../../../pure';
import { userEvent } from '../..';
import { createEventLogger } from '../../../test-utils';

describe('userEvent.longPress with fake timers', () => {
  beforeEach(() => {
    jest.useFakeTimers();
    jest.setSystemTime(0);
  });

  test('calls onLongPress if the delayLongPress is the default one', async () => {
    const { logEvent, events } = createEventLogger();
    const user = userEvent.setup();

    render(
      <Pressable onPress={logEvent('press')} onLongPress={logEvent('longPress')}>
        <Text>press me</Text>
      </Pressable>,
    );
    await user.longPress(screen.getByText('press me'));

    expect(events).toMatchSnapshot();
  });

  test('calls onLongPress when duration is greater than specified delayLongPress', async () => {
    const mockOnLongPress = jest.fn();
    const mockOnPress = jest.fn();
    const user = userEvent.setup();

    render(
      <Pressable delayLongPress={800} onLongPress={mockOnLongPress} onPress={mockOnPress}>
        <Text>press me</Text>
      </Pressable>,
    );

    await user.longPress(screen.getByText('press me'), {
      duration: 1000,
    });

    expect(mockOnLongPress).toHaveBeenCalled();
    expect(mockOnPress).not.toHaveBeenCalled();
  });

  test('does not calls onLongPress when duration is lesser than specified delayLongPress', async () => {
    const mockOnLongPress = jest.fn();
    const mockOnPress = jest.fn();
    const user = userEvent.setup();

    render(
      <Pressable delayLongPress={1000} onLongPress={mockOnLongPress} onPress={mockOnPress}>
        <Text>press me</Text>
      </Pressable>,
    );
    await user.longPress(screen.getByText('press me'));

    expect(mockOnLongPress).not.toHaveBeenCalled();
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  test('does not calls onPress when onLongPress is called', async () => {
    const mockOnLongPress = jest.fn();
    const mockOnPress = jest.fn();
    const user = userEvent.setup();

    render(
      <Pressable onLongPress={mockOnLongPress} onPress={mockOnPress}>
        <Text>press me</Text>
      </Pressable>,
    );
    await user.longPress(screen.getByText('press me'));

    expect(mockOnLongPress).toHaveBeenCalled();
    expect(mockOnPress).not.toHaveBeenCalled();
  });

  test('longPress is accessible directly in userEvent', async () => {
    const mockOnLongPress = jest.fn();

    render(
      <Pressable onLongPress={mockOnLongPress}>
        <Text>press me</Text>
      </Pressable>,
    );

    await userEvent.longPress(screen.getByText('press me'));

    expect(mockOnLongPress).toHaveBeenCalled();
  });
});
