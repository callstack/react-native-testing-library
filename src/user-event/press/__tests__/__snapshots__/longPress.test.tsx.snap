// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userEvent.longPress with fake timers calls onLongPress if the delayLongPress is the default one 1`] = `
[
  {
    "name": "longPress",
    "payload": {
      "currentTarget": {
        "measure": [Function],
      },
      "dispatchConfig": {
        "registrationName": "onResponderGrant",
      },
      "nativeEvent": {
        "changedTouches": [],
        "identifier": 0,
        "locationX": 0,
        "locationY": 0,
        "pageX": 0,
        "pageY": 0,
        "target": 0,
        "timestamp": 0,
        "touches": [],
      },
      "persist": [Function],
      "target": {},
    },
  },
]
`;
